compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -536,9 +547,162 @@ private STToken processNumericLiteral(int startChar) {
 
     /**
      * <p>
-     * Process and returns a hex integer literal.
+     * Process and returns a decimal floating point literal.
+     * </p>
+     * <code>
+     * DecimalFloatingPointNumber :=
+     *    DecimalNumber Exponent [FloatingPointTypeSuffix]
+     *    | DottedDecimalNumber [Exponent] [FloatingPointTypeSuffix]
+     *    | DecimalNumber FloatingPointTypeSuffix
+     * <br/>
+     * DottedDecimalNumber := DecimalNumber . Digit* | . Digit+
+     * <br/>
+     * FloatingPointTypeSuffix := DecimalTypeSuffix | FloatTypeSuffix
+     * <br/>
+     * DecimalTypeSuffix := d | D
+     * <br/>
+     * FloatTypeSuffix :=  f | F
+     * </code>
+     *
+     * @param isDotStart Whether process starts from a dot
+     * @return The decimal floating point literal.
+     */
+    private STToken processDecimalFloatLiteral(boolean isDotStart) {
+        int nextChar = peek();
+
+        // Direct process to the dot switch case
+        if (isDotStart) {
+            nextChar = LexerTerminals.DOT;
+        }
+
+        switch (nextChar) {
+            case LexerTerminals.DOT:
+                if (!isDotStart) { // Advance only if it is not already validated
+                    reader.advance();
+                }
+                nextChar = peek();
+                while (true) {
+                    if (isDigit(nextChar)) {
+                        reader.advance();
+                        nextChar = peek();
+                        continue;
+                    }
+                    break;
+                }
+                switch (nextChar) {
+                    case 'e':
+                    case 'E':
+                        return processExponent(false);
+                    case 'f':
+                    case 'F':
+                    case 'd':
+                    case 'D':
+                        return parseFloatingPointTypeSuffix();
+                }
```

Comment:
This code block looks like a duplicate of L#602-L#610. Can we use a single switch-case for both scenarios?